File: backend/src/main/java/br/ufpr/sept/so2/modules/iam/application/ports/UsuarioRepository.java
package br.ufpr.sept.so2.modules.iam.application.ports;

import br.ufpr.sept.so2.modules.iam.domain.IdentificadorLogin;
import br.ufpr.sept.so2.modules.iam.domain.Usuario;

import java.util.Optional;
import java.util.UUID;

public interface UsuarioRepository {

    Usuario save(Usuario usuario);

    Optional<Usuario> findById(UUID id);

    Optional<Usuario> findByIdentificador(IdentificadorLogin identificador);

    Optional<Usuario> findByEmail(String email);
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/iam/application/ports/UsuarioRepository.kt
package br.ufpr.sept.so2.modules.iam.application.ports

import br.ufpr.sept.so2.modules.iam.domain.IdentificadorLogin
import br.ufpr.sept.so2.modules.iam.domain.Usuario
import java.util.Optional
import java.util.UUID

interface UsuarioRepository {
    fun save(usuario: Usuario): Usuario

    fun findById(id: UUID): Optional<Usuario>

    fun findByIdentificador(identificador: IdentificadorLogin): Optional<Usuario>

    fun findByEmail(email: String): Optional<Usuario>
}
